Update definitions for redux-persist plugins

LKay opened the pull request:
Please fill in this template.

- [X] Make your PR against the `master` branch.
- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `tsc` without errors.
- [X] Run `npm run lint package-name` if a `tslint.json` is present.

If changing an existing definition:
- [] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [X] Increase the version number in the header if appropriate.
- [X] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [X] Delete the package's directory.
- [X] Add it to `notNeededPackages.json`.

LKay commented:
Waiting for the https://github.com/rt2zz/redux-persist/pull/282 to be merged.

LKay commented:
Is there and problem with building script right now?

Created definitions for new versions of the libraries and move old ones to `v0.1` directory. I also updated `tsconfig.json` providing `paths` for the old typings and builder seems to ignore `index.d.ts` even if it's defined under `files` property.

/cc @andy-ms

andy-ms commented:
We don't support older minor versions. Just delete `v0.1`.

mhegazy commented:
Thank you for your contribution. 

Can you please fix the failing Travis CI tests and update the PR!

mhegazy commented:
still failing.

LKay commented:
I know, there is problem with redux-persist typing that this typings depend on. I already made the fix which got merged yesterday. As soon as it's released the build should be fixed.

paulvanbrenk commented:
Closing, can you fix the merge conflict and the CI build.

LKay commented:
@paulvanbrenk The new version of `redux-persist` fixing mentioned errors got released on Friday. I'm fixing the build right now, and will commit changes soon.

LKay commented:
@paulvanbrenk Should be good to go now. Tests and CI build seem to pass. Can you reopen this PR?

andy-ms commented on notNeededPackages.json:
```diff
@@ -355,10 +355,64 @@
             "asOfVersion": "2.0.0"
         },
         {
+            "libraryName": "ecmarkup",
```
You only meant to add one package here, right? Git must have gotten confused due to #15264.

LKay commented:
Yeah, I wanted to ad 2 packages there. Git messed something up during merge with master branch. Will fix that tomorrow morning (JST time).

LKay commented:
@andy-ms I recreated a `notNeededPackages.json` based on copy-pasted file from master branch and added two records for the packages I removed, but still github messes something up in diff. Should be fine, though.

andy-ms commented:
If github shows a diff there's probably something real happening. Try rebasing?

LKay commented:
@andy-ms Ok, this should do the job. Only that editor auto trimmed spaces at the end of lines for one record.

paulvanbrenk commented:
Thanks!
